src/main/java/com/google/firebase/auth/FirebaseUserManager.java

```diff
@@ -384,12 +389,19 @@ void deleteSamlProviderConfig(String providerId) throws FirebaseAuthException {
     sendRequest("DELETE", url, null, GenericJson.class);
   }
 
-  private static String generateMask(Map<String, Object> properties) {
-    // This implementation does not currently handle the case of nested properties. This is fine
-    // since we do not currently generate masks for any properties with nested values. When it
-    // comes time to implement this, we can check if a property has nested properties by checking
-    // if it is an instance of the Map class.
-    return Joiner.on(",").join(ImmutableSortedSet.copyOf(properties.keySet()));
+  private static ImmutableSortedSet<String> generateMask(Map<String, Object> properties) {
```

Comment:
Declare return type to be `Set<String>` for extra flexibility.

Reply:
Making the return type `Set<String>` does not give any extra flexibility (since this is not an interface). If this were an input parameter, we would certainly want to accept `Set` or even `Collection` instead of `ImmutableSortedSet`.

In Java, it's best to make the return type as specific as possible. In this case, indicating that the return value is both immutable and sorted is important. See go/java-practices/methods#type_generality.

Reply:
We spoke and decided to change it to `Set` in order to avoid Guava from leaking into the method signatures.